### PR TITLE
Add native tools on machine doc

### PR DESCRIPTION
**Pull request by jonfortescue**
Doc I wrote on Friday but forgot to post as a PR for some reason. Realized I should have a doc explaining migration that I can link to in the email so that's what this is.

### CODE REVIEW

File: Documentation/NativeToolsOnMachine.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+This document only applies to Windows machines as Linux machines already come with native tools installed via package manager.
+
+## Steps
+
+1.  Before calling `InitializeNativeTools`, `init-tools-native.ps1`, or `eng/common`'s `build.ps1` in PowerShell,
+    set `$env:NativeToolsOnMachine = $True`. Alternatively, simply set it as an environment variable when calling
+    a template such as `jobs.yml` in YAML.
+2.  Modify your `global.json`'s `native-tools` section to change the version of your tools to one of the following values:
+    - `latest` (e.g. `"cmake": "latest"`) &ndash; Grabs the latest version of the tool on the machine; this should be what you use in the majority of cases
+    - Major version (e.g. `"cmake": "3"`) &ndash; Grabs the latest minor & patch version of a particular major version on the machine; this is useful artifacts you want to pin to a specific major version
```
Comment (MattGal) [inline Documentation/NativeToolsOnMachine.md:15]: nit:  Missing a word, this is useful **for** artifacts?

File: Documentation/NativeToolsOnMachine.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    - Minor version (e.g. `"python": "3.10"`) &ndash; Grabs the latest patch version of a particular minor version on the machine; this is useful for artifacts you want to pin to a specific minor version (such as Python)
```
Comment (MattGal) [inline Documentation/NativeToolsOnMachine.md:16]: Do we actually provide multiple pythons ?  WHy?

File: Documentation/NativeToolsOnMachine.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+3.  Adjust any usage of the artifacts on the machines in your scripts. The artifacts you specify in your `global.json` are promoted to the path,
+    so in general simply calling `cmake` will work. However, if you need the specific locations of tools, `InitializeNativeTools` will return those to
```
Comment (MattGal) [inline Documentation/NativeToolsOnMachine.md:18]: Consider combining lines here, this may look odd on some column widths.

**Review by MattGal — APPROVED** [review]

File: Documentation/NativeToolsOnMachine.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+# Using Native Tools on Machine
+
+The replacement for [native tools bootstrapping](NativeToolsBootstrapping.md) is having all tools needed come pre-installed on
```
Comment (ChadNedzlek) [inline Documentation/NativeToolsOnMachine.md:3]: If this is a replacement for that... shouldn't we delete that?

Comment (MattGal) [reply Documentation/NativeToolsOnMachine.md:3]: I think we have to phase the users over to the new images that have built-in native dependencies first.

File: Documentation/NativeToolsOnMachine.md (hunk 2 above)
Comment (jonfortescue) [reply Documentation/NativeToolsOnMachine.md:16]: No, this was just an example of a minor version specification. Python's the only case I could think of for that lol, but hypothetically there are other cases where you'd want to pin a minor version.

File: Documentation/NativeToolsOnMachine.md (hunk 3 above)
Comment (jonfortescue) [reply Documentation/NativeToolsOnMachine.md:18]: I do separate lines so it's easy to pinpoint comments in a PR; it all looks the same when it's rendered in markdown

File: Documentation/NativeToolsOnMachine.md (hunk 4 above)
Comment (jonfortescue) [reply Documentation/NativeToolsOnMachine.md:3]: What Matt said, yeah

Comment (ChadNedzlek) [reply Documentation/NativeToolsOnMachine.md:3]: Makes sense!

**Review by MattGal — APPROVED** [review]

**Pull request merged**